### PR TITLE
[bitnami/airflow] adds support for retrieving the password for external Postgres and Redis from an existing secret

### PR DESCRIPTION
**Pull request by chgl**
**Description of the change**

This adds an option `existingSecret` to `externalDatabase` and `externalRedis` which can be used to specify the name of an existing secret which contains the passwords for both.

**Benefits**

Externalize the password management to an existing secret resource which, for example, may be created by Bitnami's sealed secret operator.

**Possible drawbacks**

**Applicable issues**

**Additional information**

I've also replaced the word "Kafka" with "Airflow" in the README.

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/airflow/README.md [hunk 1]
```diff
@@ -256,6 +258,41 @@ airflow.clonePluginsFromGit.branch=v1.0.9-branch
...
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: all-my-secrets
+type: Opaque
+data:
+  airflow-password: "Smo1QTJLdGxXMg=="
+  airflow-fernetKey: "YVRZeVJVWnlXbU4wY1dOalVrdE1SV3cxWWtKeFIzWkVRVTVrVjNaTFR6WT0="
+  db-password: "cG9zdGdyZXMK"
```
Comment (sameersbn) [inline bitnami/airflow/README.md]: The stable/postgresql chart uses `postgresql-password` for the user password.If we use the same key then user can simply specify the secret of an existing stable/postgresql chart deployment

```suggestion
  postgresql-password: "cG9zdGdyZXMK"
```

File: bitnami/airflow/templates/deployment-scheduler.yaml [hunk 2]
```diff
@@ -128,6 +128,9 @@ spec:
                 {{- if .Values.postgresql.enabled }}
                   name: {{ template "airflow.postgresql.fullname" . }}
                   key: postgresql-password
+                {{- else if .Values.externalDatabase.existingSecret}}
+                  name: {{ .Values.externalDatabase.existingSecret }}
+                  key: db-password
```
Comment (sameersbn) [inline bitnami/airflow/templates/deployment-scheduler.yaml]: I suggest adding a helper function like the one below:

```golang
{{/*
Get the postgresql credentials secret.
*/}}
{{- define "airflow.postgresql.secretName" -}}
{{- if .Values.externalDatabase.existingSecret -}}
    {{- printf "%s" .Values.externalDatabase.existingSecret -}}
{{- else -}}
    {{- printf "%s" (include "airflow.postgresql.fullname" .) -}}
{{- end -}}
{{- end -}}
```

and then replacing 

```yaml
                {{- if .Values.postgresql.enabled }}
                  name: {{ template "airflow.postgresql.fullname" . }}
                  key: postgresql-password
                {{- else if .Values.externalDatabase.existingSecret}}
                  name: {{ .Values.externalDatabase.existingSecret }}
                  key: db-password
                {{- else }}
```

with

```
                {{- if .Values.postgresql.enabled }}
                  name: {{ template "airflow.postgresql.secretName" . }}
                  key: postgresql-password
                {{- else }}
```

The same should also be done for the Redis bits

File: bitnami/airflow/README.md (hunk 1 above)
Comment (chgl) [reply bitnami/airflow/README.md]: Will do! For consistency, should we rename the `db-password` key in `secret-externaldb.yaml` to `postgresql-password` as well?

File: bitnami/airflow/templates/deployment-scheduler.yaml (hunk 2 above)
Comment (chgl) [reply bitnami/airflow/templates/deployment-scheduler.yaml]: I'm now determining the `secretKeyRef.name` entirely in the template:

https://github.com/bitnami/charts/pull/1917/commits/2ca88f1b472b796e506d9bef21486cb6aaa79b3f#diff-404f4950f79b92ad20c70deadbe1bb17

File: bitnami/airflow/templates/secret-externaldb.yaml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {{- if not .Values.postgresql.enabled }}
+{{- if not .Values.externalDatabase.existingSecret }}
```
Comment (sameersbn) [inline bitnami/airflow/templates/secret-externaldb.yaml]: we can combine the above lines to 

```
{{- if and (not .Values.postgresql.enabled) (not .Values.externalDatabase.existingSecret) }}
```

File: bitnami/airflow/templates/secret-externalredis.yaml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {{- if not .Values.redis.enabled }}
+{{- if not .Values.externalRedis.existingSecret }}
```
Comment (sameersbn) [inline bitnami/airflow/templates/secret-externalredis.yaml]: we can combine this as well.

**Pull request merged**